zerver/actions/streams.py

```diff
@@ -750,7 +750,7 @@ def send_subscription_remove_events(
                 stream.recipient_id for stream in streams_by_user[user_profile.id]
             ],
         }
-        queue_json_publish("deferred_work", event)
+        queue_event_on_commit("deferred_work", event)
```

Comment:
Since we call `queue_json_publish` in a loop, we cannot use `transaction.on_commit` directly. I tried using the method suggested in commit message in 7657cb4a0f, but ruff does not allow this by saying something on the lines lambda function is executed directly. So, just added this function to handle this how we handle `send_event` calls.

Reply:
Seems very reasonable! We'll probably want to look at a broader sweep to use that; may be worth a #backend conversation.